contracts/FlowIDTableStaking.cdc

```diff
@@ -408,14 +402,24 @@ pub contract FlowIDTableStaking {
         /// Stake tokens that are in the tokensUnstaked bucket 
         pub fun stakeUnstakedTokens(amount: UFix64) {
 
-            if amount > 0.0 {
+            let nodeRecord = FlowIDTableStaking.borrowNodeRecord(self.id)
 
-                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(self.id)
+            var remainingAmount = amount
+
+            if remainingAmount <= nodeRecord.tokensRequestedToUnstake {
+                nodeRecord.tokensRequestedToUnstake = nodeRecord.tokensRequestedToUnstake - remainingAmount
+                remainingAmount = 0.0
+            } else if remainingAmount > nodeRecord.tokensRequestedToUnstake {
+                remainingAmount = remainingAmount - nodeRecord.tokensRequestedToUnstake
+                nodeRecord.tokensRequestedToUnstake = 0.0
+            }
+
+            if remainingAmount > 0.0 {
 
                 /// Add the removed tokens to tokens committed
-                nodeRecord.tokensCommitted.deposit(from: <-nodeRecord.tokensUnstaked.withdraw(amount: amount))
+                nodeRecord.tokensCommitted.deposit(from: <-nodeRecord.tokensUnstaked.withdraw(amount: remainingAmount))
 
-                emit TokensCommitted(nodeID: nodeRecord.id, amount: amount)
+                emit TokensCommitted(nodeID: nodeRecord.id, amount: remainingAmount)
             }
```

Comment:
I guess we don't have any events related to unstaking, it's just seting the amount to 0.0

Reply:
yeah the event for unstaking doesn't happen until the actual tokens are unstaked